controllers/validator.go

```diff
@@ -63,6 +64,17 @@ func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error)
 	if _, err := getResticResourceReqs(&dpa); err != nil {
 		return false, err
 	}
+
+	// validate non-admin enable and tech-preview-ack
+	if !r.checkNonAdminEnabled(&dpa) {
+		if dpa.Spec.NonAdmin != nil &&
+			dpa.Spec.NonAdmin.Enable != nil &&
+			boolptr.IsSetToTrue(dpa.Spec.NonAdmin.Enable) &&
```

Comment:
is not this unnecessary?

we already check that `dpa.Spec.NonAdmin.Enable` is not `nil`

Reply:
would not be simple to do not change `checkNonAdminEnabled` function and just add check here?
```go
if r.checkNonAdminEnabled(&dpa) {
    if !(dpa.Spec.UnsupportedOverrides[oadpv1alpha1.TechPreviewAck] == TrueVal) {
        return false, errors.New("in order to enable the non-admin feature please set dpa.spec.unsupportedOverrides[tech-preview-ack]: true")
}
```
this way, when it GA, we just need to remove these lines here in validator?

Reply:
actually, I think is best to add this only in non admin controller

if I enable both unsupported override and enable dpa field, non admin deployment will appear in cluster

if later, I remove the unsupported, validation here will break, right? but this happens before non admin reconcile, which would let no admin deployment still running, right?

Reply:
@mateusoliveira43 I decoupled non-admin enable and tech preview ack flag in the latest commit, PTAL, hope that resolves everything, Thanks !

Reply:
I tested with `make deploy-olm`

semantically, I believe is correct to put unsupportedOverrides validation in validator (it is a DPA validation) 

But the scenario where user deploys non-admin correctly, and later changes unsupportedOverrides, makes reconcile fail for DPA, but does not delete non-admin deployment. Would be safer to put all validation in non-admin controller? It can also confuse users I think, as the DPA message will tell `in order to enable the non-admin feature please ...`, but it will be still enabled

Reply:
Ok maybe we just re-frame the DPA message to `in order to enable/disable the non-admin feature please...` WDYT ?

Reply:
makes sense to me!

Reply:
Done, updated the PR in f7e2b1200ff8d53359b7585a1f087b87f9d0bccd